src/Testcontainers.Db2/Db2Builder.cs

```diff
@@ -0,0 +1,202 @@
+
+    /// <summary>
+    /// Sets the Db2 database name.
+    /// </summary>
+    /// <param name="database">The Db2 database.</param>
+    /// <returns>A configured instance of <see cref="Db2Builder" />.</returns>
+    public Db2Builder WithDatabase(string database)
+    {
+        return Merge(DockerResourceConfiguration, new Db2Configuration(database: database))
+            .WithEnvironment("DBNAME", database);
+    }
+
+    /// <summary>
+    /// Sets the Db2 username.
+    /// </summary>
+    /// <param name="username">The Db2 username.</param>
+    /// <returns>A configured instance of <see cref="Db2Builder" />.</returns>
+    public Db2Builder WithUsername(string username)
+    {
+        return Merge(DockerResourceConfiguration, new Db2Configuration(username: username))
+            .WithEnvironment("DB2INSTANCE", username);
+    }
+
+    /// <summary>
+    /// Sets the Db2 password.
+    /// </summary>
+    /// <param name="password">The Db2 password.</param>
+    /// <returns>A configured instance of <see cref="Db2Builder" />.</returns>
+    public Db2Builder WithPassword(string password)
+    {
+        return Merge(DockerResourceConfiguration, new Db2Configuration(password: password))
+            .WithEnvironment("DB2INST1_PASSWORD", password);
+    }
+
+    /// <summary>
+    /// Sets the Db2 archive logs.
+    /// </summary>
+    /// <param name="archiveLogs">The Db2 archive logs setting.</param>
+    /// <returns>A configured instance of <see cref="Db2Builder" />.</returns>
+    public Db2Builder WithArchiveLogs(bool archiveLogs)
+    {
+        return Merge(DockerResourceConfiguration, new Db2Configuration(archiveLogs: archiveLogs))
+            .WithEnvironment("ARCHIVE_LOGS", archiveLogs.ToString());
+    }
+
+    /// <summary>
+    /// Sets the Db2 autoconfig setting.
+    /// </summary>
+    /// <param name="autoConfig">The Db2 autoconfig setting.</param>
+    /// <returns>A configured instance of <see cref="Db2Builder" />.</returns>
+    public Db2Builder WithAutoconfig(bool autoConfig)
+    {
+        return Merge(DockerResourceConfiguration, new Db2Configuration(autoConfig: autoConfig))
+            .WithEnvironment("AUTOCONFIG", autoConfig.ToString());
+    }
+
+    /// <summary>
+    /// Accepts the Db2 license agreement.
+    /// </summary>
+    /// <returns>A configured instance of <see cref="Db2Builder" />.</returns>
+    public Db2Builder WithLicenseAgreement()
+    {
+        return Merge(DockerResourceConfiguration, new Db2Configuration(licenseAgreement: DefaultLicenseAgreement))
+            .WithEnvironment("LICENSE", DefaultLicenseAgreement);
+    }
+
+    /// <summary>
+    /// Maps the database to memory.
+    /// </summary>
+    /// <returns>A configured instance of <see cref="Db2Builder" />.</returns>
+    public Db2Builder WithInMemoryDatabase()
+    {
+        return Merge(DockerResourceConfiguration, new Db2Configuration(licenseAgreement: DefaultInMemoryDatabasePath))
```

Comment:
This line passes the value to the wrong argument.

Reply:
Will adjust